flocker/node/agents/cinder.py

```diff
@@ -138,49 +175,132 @@ def create_volume(self, dataset_id, size):
             CLUSTER_ID_LABEL: unicode(self.cluster_id),
             DATASET_ID_LABEL: unicode(dataset_id),
         }
-        # We supply metadata here and it'll be included in the returned cinder
-        # volume record, but it'll be lost by Rackspace, so...
-        requested_volume = self.volume_manager.create(
-            size=Byte(size).to_GB().value,
-            metadata=metadata,
+        action_type = u"blockdevice:cinder:create_volume"
+        with start_action(action_type=action_type):
+            requested_volume = self.cinder_volume_manager.create(
+                size=Byte(size).to_GB().value,
+                metadata=metadata,
+            )
+            Message.new(blockdevice_id=requested_volume.id).write()
+            wait_for_volume(
+                volume_manager=self.cinder_volume_manager,
+                expected_volume=requested_volume,
+            )
+        return _blockdevicevolume_from_cinder_volume(
+            cinder_volume=requested_volume,
```

Comment:
It isn't clear why this is now the `requeseted_volume` instead of `created_volume`. I guess the attributes that are inspected don't change?